spring-cloud-function-context/src/main/java/org/springframework/cloud/function/context/ContextFunctionCatalogAutoConfiguration.java

```diff
@@ -367,16 +367,38 @@ private ProxyConsumer wrapConsumer(Consumer<Flux<?>> consumer,
 			return wrapped;
 		}
 
-		private Class<?> findType(RootBeanDefinition definition, int index) {
-			StandardMethodMetadata source = (StandardMethodMetadata) definition
-					.getSource();
+		private Class<?> findType(AbstractBeanDefinition definition, int index) {
+			Object source = definition.getSource();
 			Type param;
 			if (source instanceof StandardMethodMetadata) {
 				ParameterizedType type;
-				type = (ParameterizedType) (source.getIntrospectedMethod()
-						.getGenericReturnType());
-				type = (ParameterizedType) type.getActualTypeArguments()[index];
-				param = type.getActualTypeArguments()[0];
+				type = (ParameterizedType) ((StandardMethodMetadata) source).getIntrospectedMethod()
+						.getGenericReturnType();
+				Type typeArgumentAtIndex = type.getActualTypeArguments()[index];
```

Comment:
This statement and the `if-else` block below is repeated. Please consider extract it into a method.